memcached.c

```diff
@@ -4191,6 +4287,837 @@ static inline void process_get_command(conn *c, token_t *tokens, size_t ntokens,
     }
 }
 
+// slow snprintf for debugging purposes.
+static void process_meta_command(conn *c, token_t *tokens, const size_t ntokens) {
+    assert(c != NULL);
+
+    if (tokens[KEY_TOKEN].length > KEY_MAX_LENGTH) {
+        out_string(c, "CLIENT_ERROR bad command line format");
+        return;
+    }
+
+    char *key = tokens[KEY_TOKEN].value;
+    size_t nkey = tokens[KEY_TOKEN].length;
+
+    item *it = limited_get(key, nkey, c, 0, false, DONT_UPDATE);
+    if (it) {
+        size_t total = 0;
+        size_t ret;
+        // similar to out_string().
+        memcpy(c->wbuf, "ME ", 3);
+        total += 3;
+        memcpy(c->wbuf + total, ITEM_key(it), it->nkey);
+        total += it->nkey;
+        c->wbuf[total] = ' ';
+        total++;
+
+        ret = snprintf(c->wbuf + total, c->wsize - (it->nkey + 12),
+                "exp=%d la=%llu cas=%llu fetch=%s cls=%u size=%lu\r\nEN\r\n",
+                (it->exptime == 0) ? -1 : (current_time - it->exptime),
+                (unsigned long long)(current_time - it->time),
+                (unsigned long long)ITEM_get_cas(it),
+                (it->it_flags & ITEM_FETCHED) ? "yes" : "no",
+                ITEM_clsid(it),
+                (unsigned long) ITEM_ntotal(it));
+
+        item_remove(it);
+        c->wbytes = total + ret;
+        c->wcurr = c->wbuf;
+        conn_set_state(c, conn_write);
+        c->write_and_go = conn_new_cmd;
+    } else {
+        out_string(c, "EN");
+    }
+    pthread_mutex_lock(&c->thread->stats.mutex);
+    c->thread->stats.mget_cmds++;
```

Comment:
stats only get updated for the `me` command, shouldn't this happen in the `process_mget_command` method?

Reply:
I should change it from mget -> meta. mg/ms/md count as get/set/delete commands. I believe that's the right approach because otherwise adopting the commands will screw everyone's metrics. I left a note about needing new stats for specific sub-actions like autoviv or mark-stale or etc.

Reply:
fixed this instance. adding rest of counters is low prio atm.